app/main.py

```diff
@@ -1,6 +1,18 @@
+from functools import wraps
 from typing import Callable
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    @wraps(func)
+    def inner(*args, **kwargs) -> None:
+        key = (args, tuple(kwargs.items()))
+        if key not in cache_dict:
+            print("Calculating new result")
+            result = func(*args, **kwargs)
+            cache_dict[key] = result
```

Comment:
Please pass the result of the function execution directly to dict, instead of creating a separate `result ` variable.